File: evolab-backend/domain/src/main/kotlin/com/example/evolab/domain/project/Project.kt
package com.example.evolab.domain.project

import com.example.evolab.domain.evolution.EvolutionStatus
import org.w3c.dom.Text
import java.time.Instant

data class Project(
    val id: Int,
    val userId: Int,
    val configId: Int?,
    val name: String,
    val description: String?,
    val initialProgram: String?,
    val evaluatorCode: String?,
    val status: EvolutionStatus,
    val createdAt: Instant,
)

File: evolab-frontend/EvoLabAsAService/src/types/project.ts
import { EvolutionStatus } from './evolution';

export interface Project {
    id: number;
    userId: number;
    configId: number | null;
    name: string;
    description: string | null;
    initialProgram: string | null;
    evaluatorCode: string | null;
    status: EvolutionStatus;
    createdAt: string; // ISO 8601 Instant
}

export interface CreateProjectInput {
    name: string;
    description: string | null;
    configId: number | null;
    initialProgram: string | null;
    evaluatorCode: string | null;
}

export interface UpdateProjectDetailsInput {
    name: string | null;
    description: string | null;
    configId: number | null;
    initialProgram: string | null;
    evaluatorCode: string | null;
}

export interface UpdateProjectStatusInput {
    status: EvolutionStatus;
}

